docs/guides/src/main/server/config-mainaspects.adoc

```diff
@@ -0,0 +1,128 @@
+<#import "/templates/guide.adoc" as template>
+<#import "/templates/kc.adoc" as kc>
+
+<@template.guide
+title="Core concepts for configuring Keycloak"
+summary="Understand main aspects around Keycloaks configuration">
+
+With the underlying Quarkus distribution, configuring Keycloak got a lot easier. There are some main aspects to understand, though. This guide will help you to understand the core concepts around the configuration of Keycloak.
+
+== Distinction between runtime- and buildtime options
+Configuration options are divided in two categories:
+
+- Runtime Options are options that can be set at runtime when starting the server
+
+- Buildtime Options are options that can only be set when configuring the server through the `build` command
+
+This distinction is used to optimize Keycloak for a good startup time and memory footprint.
+
+The `build` command starts a build process during which Keycloaks configuration gets baked into the jar, and thus optimizes the resulting Keycloak server image.
+
+Under the covers, Keycloak leverages Quarkus' "reaugmentation" phase to produce a fully optimized Keycloak server image including all settings. In this phase, multiple initialization steps (e.g. loading providers, switching features on or off, setting up the db connection) are executed and baked into the Keycloak server image. This saves a good amount of time when starting up.
+
+We recommend you to use the `build` step to get the best behaviour you can out of Keycloak.
+
+To summarize, this distinction and the underlying concepts provide a way to tailor a fully optimized Keycloak server image for the specific environment Keycloak runs in, and is essential to achieve fast startup time and low memory footprint.
+
+=== List of buildtime options
+
+<#list ctx.options.categories as category>
```

Comment:
I don't think having options available to the build command should be here. From my PoV, guides should focus on explaining concepts and be more consice. 

We can just update the all-config page with the information about whether a option requires a build. As well as teach people how to query if a property requires a build (e.g.: help).